src/xSTIR/pSTIR/STIR.py

```diff
@@ -1627,22 +1627,40 @@ def get_time_at_which_num_prompts_exceeds_threshold(self, threshold):
 
 class PoissonNoiseGenerator(object):
     """
-    Class that generates Poisson noise.
+    Generates noise realisations according to Poisson statistics but allowing for scaling.
+
+    A scaling_factor is used to multiply the input data before generating
+    the Poisson random number. This means that a scaling_factor larger than 1
+    will result in less noisy data.
+
+    If preserve_mean=false, the mean of the output data will
+    be equal to <tt>scaling_factor*mean_of_input</tt>, otherwise it
+    will be equal to mean_of_input, but then the output is no longer Poisson
+    distributed.
     """
 
     def __init__(self, scaling_factor=1.0, preserve_mean=False):
         self.name = "PoissonNoiseGenerator"
         self.handle = pystir.cSTIR_createPoissonNoiseGenerator(scaling_factor, preserve_mean)
         check_status(self.handle)
-        self.scale = scaling_factor
-
-    @property
-    def scaling_factor(self):
-        return self.scale
+        self.scaling_factor = scaling_factor
+        self.preserve_mean = preserve_mean
+        self.output_handle = None
```

Comment:
As I mentioned, as these are accessible, a user will expect that they can be set after creation, but that will give an incorrect result.

I thought this could be easiest to handle by moving the call to `pystir.cSTIR_createPoissonNoiseGenerator` to `process()`, but then I think we run into difficulties with setting `seed`.. So, either remove these members, or make them read-only (not sure how to do that in Python).

Reply:
removed